### PR TITLE
[W4.4b][W10-A1]Joshua Lee Kai Sheng

### PR DESCRIPTION
**Pull request by jlks96**
Completed [LO-ImplementClass]

### CODE REVIEW

File: doc/UserGuide.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ Format: `help`
  
 ### Adding a person: `add`
 Adds a person to the address book<br>
-Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...` 
+Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/BLOCK, STREET, UNIT, POSTAL_CODE [t/TAG]...`
```
Comment (okkhoy) [inline doc/UserGuide.md:38]: Good to see you have updated the user guide and the examples.

File: src/seedu/addressbook/commands/AddCommand.java [hunk 2]
```diff
@@ -23,9 +23,9 @@
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a person to the address book. "
             + "Contact details can be marked private by prepending 'p' to the prefix.\n"
-            + "Parameters: NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS  [t/TAG]...\n"
+            + "Parameters: NAME [p]p/PHONE [p]e/EMAIL [p]a/BLOCK, STREET, UNIT, POSTAL_CODE  [t/TAG]...\n"
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/AddCommand.java:26]: and the help :+1:

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -12,7 +12,10 @@
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
-    public final String value;
+    public final Block block;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java:15]: Try to make it private. In general, start off with private members, then use protected or public only if absolutely necessary.

File: src/seedu/addressbook/data/person/Block.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package seedu.addressbook.data.person;
+
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java:2]: This class and its non-trivial methods should have header comments. 
If not, developers who write client code for this class will not know how to use it.

File: src/seedu/addressbook/data/person/Block.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+public class Block {
+
+    String blockNumber;
+
+    public Block(String blockNumber) {
+        this.blockNumber = blockNumber;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java:8]: You can try to move the validation logic into these classes.

**Comment by okkhoy** [comment]
@jlks96 
This is a good attempt!
Some comments added. Please ack & close the PR after reading comments.

**Comment by jlks96** [comment]
Acknowledged and thanks!

**Pull request closed without merging**